deploy/deploy_test.go

```diff
@@ -268,65 +358,82 @@ func TestKindSpec(t *testing.T) {
 			Name:    "test",
 			Version: "v0.15.0",
 		},
-		vExecer: exec.NewFakeExecerWithIO(vOut, vOut, exec.Response{Stdout: "kind v0.14.0 go1.18.2 linux/amd64"}),
+		resp: []fexec.Response{
+			{Cmd: "kind", Args: []string{"version"}, Stdout: "kind v0.14.0 go1.18.2 linux/amd64", Err: "kind version check failed: got v0.14.0, want v0.15.0"},
+		},
 		wantErr: "kind version check failed: got v0.14.0, want v0.15.0",
 	}, {
 		desc: "failed kind version exec fail",
 		k: &KindSpec{
 			Name:    "test",
 			Version: "v0.15.0",
 		},
-		vExecer: exec.NewFakeExecerWithIO(vOut, vOut, exec.Response{Err: fmt.Errorf("exec error")}),
+		resp: []fexec.Response{
+			{Cmd: "kind", Args: []string{"version"}, Err: "failed to get kind version: exec error"},
+		},
 		wantErr: "failed to get kind version: exec error",
 	}, {
 		desc: "failed kind version parse",
 		k: &KindSpec{
 			Name:    "test",
 			Version: "v0.15.0",
 		},
-		vExecer: exec.NewFakeExecerWithIO(vOut, vOut, exec.Response{Stdout: "kind "}),
+		resp: []fexec.Response{
+			{Cmd: "kind", Args: []string{"version"}, Stdout: "kind ", Err: "failed to parse kind version from"},
```

Comment:
no need to error here, should happen from the result

Reply:
Done